File: ALittle/Module/ALittleIDE/Script/UIEdit/IDEEditCore/IDESetting/TextAreaS.lua
-- ALittle Generate Lua
module("ALittleIDE", package.seeall)

local ___thispath = select('1', ...):match("(.+[/\\]).+$") or ""
local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittleIDE.TextS, " extends class:ALittleIDE.TextS is nil")
TextAreaS = Lua.Class(ALittleIDE.TextS, "ALittleIDE.TextAreaS")

function TextAreaS:Ctor(user_info, tab_child, tree_logic)
	___rawset(self, "_layer_name", "ide_setting_textarea")
end

function TextAreaS:LoadNatureBase()
	TextS.LoadNatureBase(self)
	self:LoadTypeSelectData("halign", g_IDEEnum.HALIGN_type)
	self:LoadTypeSelectData("valign", g_IDEEnum.VALIGN_type)
end

function TextAreaS:HandleHalignSELECT_CHANGE(event)
	local list = g_IDEEnum.HALIGN_rtype
	self:TypeSelectChange("halign", list, false)
end

function TextAreaS:HandleValignSELECT_CHANGE(event)
	local list = g_IDEEnum.VALIGN_rtype
	self:TypeSelectChange("valign", list, false)
end

File: ALittle/Module/ALittleIDE/JSScript/UIEdit/IDEEditCore/IDESetting/TextAreaS.js
{
if (typeof ALittleIDE === "undefined") window.ALittleIDE = {};


if (ALittleIDE.TextS === undefined) throw new Error(" extends class:ALittleIDE.TextS is undefined");
ALittleIDE.TextAreaS = JavaScript.Class(ALittleIDE.TextS, {
	Ctor : function(user_info, tab_child, tree_logic) {
		this._layer_name = "ide_setting_textarea";
	},
	LoadNatureBase : function() {
		ALittleIDE.TextS.LoadNatureBase.call(this);
		this.LoadTypeSelectData("halign", ALittleIDE.g_IDEEnum.HALIGN_type);
		this.LoadTypeSelectData("valign", ALittleIDE.g_IDEEnum.VALIGN_type);
	},
	HandleHalignSELECT_CHANGE : function(event) {
		let list = ALittleIDE.g_IDEEnum.HALIGN_rtype;
		this.TypeSelectChange("halign", list, false);
	},
	HandleValignSELECT_CHANGE : function(event) {
		let list = ALittleIDE.g_IDEEnum.VALIGN_rtype;
		this.TypeSelectChange("valign", list, false);
	},
}, "ALittleIDE.TextAreaS");

}